torchgeo/datasets/millionaid.py

```diff
@@ -0,0 +1,397 @@
+        cls_label = [self.class_to_idx[label] for label in files["label"]]
+        label = torch.tensor(cls_label, dtype=torch.long)
+        sample = {"image": image, "label": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_files(self, root: str) -> List[Dict[str, Any]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image, and list of labels
+        """
+        imgs_no_subcat = list(
+            glob.glob(os.path.join(root, self.split, "**", "**", "*.jpg"))
+        )
+
+        imgs_subcat = list(
+            glob.glob(os.path.join(root, self.split, "**", "**", "**", "*.jpg"))
+        )
+
+        scenes = [p.split(os.sep)[-3] for p in imgs_no_subcat] + [
+            p.split(os.sep)[-4] for p in imgs_subcat
+        ]
+
+        subcategories = ["Missing" for p in imgs_no_subcat] + [
+            p.split(os.sep)[-3] for p in imgs_subcat
+        ]
+
+        classes = [p.split(os.sep)[-2] for p in imgs_no_subcat] + [
+            p.split(os.sep)[-2] for p in imgs_subcat
+        ]
+
+        if self.task == "multi-label":
+            labels = [
+                [sc, sub, c] if sub != "Missing" else [sc, c]
+                for sc, sub, c in zip(scenes, subcategories, classes)
+            ]
+        else:
+            labels = [[c] for c in classes]
+
+        images = imgs_no_subcat + imgs_subcat
+
+        files = [dict(image=img, label=l) for img, l in zip(images, labels)]
+
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        with Image.open(path) as img:
+            array: "np.typing.NDArray[np.int_]" = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _verify(self) -> None:
+        """Checks the integrity of the dataset structure.
+
+        Returns:
+            True if the dataset directories are found, else False
+        """
+        filepath = os.path.join(self.root, self.split)
+        if os.path.isdir(filepath):
+            return
+
+        filepath = os.path.join(self.root, self.split + ".zip")
+        if os.path.isfile(filepath):
+            if self.checksum and not check_integrity(filepath, self.md5s[self.split]):
+                raise RuntimeError("Dataset found, but corrupted.")
+            extract_archive(filepath)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                "Dataset not found in `root` directory, either specify a different"
```

Comment:
We should print the root directory so it's more clear where we are looking. See CDL for an example of this.